### DISCUSSION TITLE
Potential topics to share in sciwork conference 2024

### DISCUSSION THREAD

**Discussion by yungyuc** [solvcon/modmesh#419] [General]
The [sciwork](https://sciwork.dev) community is preparing for their second conference in December 2024.  Our recent work can be organized as the technical talks.  I list what I can think of now, but there may be more topics:

1. C++/Python dual mode serializer (@tigercosmos @q40603 @ThreeMonth03)
   * issue #343
   * issue #344
   * issue #348
2. Profile to speed up the 1D Euler solver (@tigercosmos @q40603 @ThreeMonth03 @j8xixo12)
   * The work may be extended to time-series code development (conversation #380)
   * issue #135
   * conversation #345
   * issue #343
3. Qt camera control and testing in Python (@Tucchhaa)
   * issue #70
   * conversation #412
4. Combine the 1D Euler solver with Qt for in-situ analysis (@Gene0315 @j8xixo12)
   * issue #202
   * issue #330
   * issue #335
   * Extended work for any linear 1D solver is tracked with issue #355
5. Time-series code development from scratch (@j8xixo12 @tigercosmos @xingularity)
   * conversation #380
6. Unstructured body mesh of mixed elements in two- and three-dimensional space (@yungyuc)
   * issue #404
7. Use Bézier curves operations interactively and in batch (@YenPeiChen07 @Gene0315)
   * issue #238
   * issue #320
   * issue #389
8. Package Windows executable (@terrychan999)
   * issue #194
9. Building process (including dependency) for Mac and Windows

I added the names of recent contributors in each of the topics.  We don't have to do a technical talk, but it usually is a productive way to organize our know-hows and becomes a stepping stone to sharpen our skills.

We can use this conversation to discuss the plan, and create new ones when needed.

**Comment by ThreeMonth03** [comment]
It sounds like a good idea. I'm interested in the issue #343, and I will attempt to solve it as my conference topic. If the remaining time allows, I'm willing to join the discussion and development in #380.

**Reply by yungyuc** [reply]
Great.  Drafting the subject and abstract for the talk can help you plan for the development.

**Reply by ThreeMonth03** [reply]
> Great. Drafting the subject and abstract for the talk can help you plan for the development.

Thank's for your useful suggestions, but I have some questions:
1. When should I finish the draft?
2. Should the draft include the topic of time series? If no, it seems that I only can talk about the reentrancy for the serializer.
3. Sould I use Chinese/English for the oral presentation?

**Reply by yungyuc** [reply]
> Thank's for your useful suggestions, but I have some questions:
> 
> 1. When should I finish the draft?

As soon as possible.  The first draft is a starting point to discuss what to do for your presentation and code.

> 2. Should the draft include the topic of time series? If no, it seems that I only can talk about the reentrancy for the serializer.

It's up to you.  It's not a bad idea to keep it simple in the beginning.  More contents can be added after the first draft.

> 3. Sould I use Chinese/English for the oral presentation?

It's up to you.

**Reply by ThreeMonth03** [reply]
> > Thank's for your useful suggestions, but I have some questions:
> > 
> > 1. When should I finish the draft?
> 
> As soon as possible. The first draft is a starting point to discuss what to do for your presentation and code.
> 
> > 2. Should the draft include the topic of time series? If no, it seems that I only can talk about the reentrancy for the serializer.
> 
> It's up to you. It's not a bad idea to keep it simple in the beginning. More contents can be added after the first draft.
> 
> > 3. Sould I use Chinese/English for the oral presentation?
> 
> It's up to you.

Ok, I will write the draft this Friday.

**Reply by ThreeMonth03** [reply]
Know your enemy and know yourself. In the context of high-performance computing, it is crucial to identify and optimize performance bottlenecks. A C++ profiler is an effective tool for pinpointing these hotspots. In the modmesh project, I contributed to the development of a profiler, and I would like to share several key aspects of it. Here are the topics I will cover:

1. The data structure and basic principles of the profiler.
2. The statistical features of the profiler.
3. The reentrant serializer of the profiler.

**Reply by yungyuc** [reply]
@ThreeMonth03 what is the subject and what is the abstract?

An abstract without bullets will be preferred.  Abstracts of well written sentences connect the rationale more clearly.

**Reply by ThreeMonth03** [reply]
> @ThreeMonth03 what is the subject and what is the abstract?
> 
> An abstract without bullets will be preferred. Abstracts of well written sentences connect the rationale more clearly.

@yungyuc Sorry for my late reply. It seems that I cannot make a presentation because I'm busy for researching.
However, I can lead @Vincent-Lien to fulfill the serialization of the profiler.

**Reply by yungyuc** [reply]
That sounds good.  We can discuss here and in discord.

**Comment by j8xixo12** [comment]
I'm interested in time series analysis, especially its application in analyzing sensor data in aerospace field.

I’m willing to give a presentation!

I will draft abstract and subject after studying some materials about FFT, I think it will take about 1 week.

**Reply by yungyuc** [reply]
Sounds, great, @j8xixo12 .  One of the simplest form of the draft can be a comment under this thread or use a new conversation.

**Comment by Gene0315** [comment]
I think I need to finish the issue https://github.com/solvcon/modmesh/pull/391 (follow the issue https://github.com/solvcon/modmesh/issues/335) first. I should have done it long ago. And then, I want to contiune the issue https://github.com/solvcon/modmesh/issues/330 to add the NRBC treatment.
Also, I'm interested in the Bézier curves fitting. 

I have a question. Do I need to write a draft?

**Reply by yungyuc** [reply]
Writing a draft will facilitate your development.  I think you can write two for both of your topics for the 1D Euler solver and the Bezier curve fitting.  What about starting with drafting the subjects for the presentation?

You do not need to give two presentations (but you have the right to).
